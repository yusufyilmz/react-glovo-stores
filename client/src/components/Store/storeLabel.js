import React from 'react';
import { StoreLabelWrapper } from './style';
import { getRandomColor } from '../../Helpers/colorHelper';

export const StoreLabel = ({ label }) => {
    return (
        <StoreLabelWrapper
            color={getRandomColor()}>
            {label}
        </StoreLabelWrapper>
    );
}