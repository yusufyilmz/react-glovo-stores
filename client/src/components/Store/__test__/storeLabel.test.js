import React from 'react';
import { shallow } from 'enzyme';
import {StoreLabel} from '../storeLabel';

describe('STORE LABEL COMPONENT', () => {
  let wrapper; let
    props;

  beforeEach(() => {
    props = {
      label: 'label'
    };

    wrapper = shallow(
      <StoreLabel {...props} />
    );
  });


  test('renders element correctly', () => {
    expect(wrapper.find('StoreLabelWrapper').length).toBe(1);
  });

  test('renders prop correctly', () => {
    expect(wrapper.find('StoreLabelWrapper').prop('color')).toBeTruthy();
  });

  test('renders text correctly', () => {
    expect(wrapper.find('StoreLabelWrapper').text()).toBe(props.label);
  });

  // test('snapchat correctly', () => {
  //   expect(wrapper).toMatchSnapshot();
  // });
});
