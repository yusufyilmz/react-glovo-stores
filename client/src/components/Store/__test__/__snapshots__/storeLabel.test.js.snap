// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`STORE LABEL COMPONENT snapchat correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StoreLabel
    label="label"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": "label",
      "color": "rgb(251,159,138)",
    },
    "ref": null,
    "rendered": "label",
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-ifAKCX",
        "isStatic": false,
        "rules": Array [
          "
    box-shadow: 0 2px 20px 0 rgba(0,0,0,.1);
    border-radius: 100px;
    font-size: 12px;
    line-height: 15px;
    padding: .50em 10px;
    background-color: ",
          [Function],
          ";
    color: #000000;
    margin: 0 2px;
    text-decoration: none!important;
    &:hover {
      text-decoration: none!important;
      color: #000000;
    }
",
        ],
      },
      "displayName": "StoreLabelWrapper",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-ifAKCX",
      "target": "span",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "label",
        "color": "rgb(251,159,138)",
      },
      "ref": null,
      "rendered": "label",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-ifAKCX",
          "isStatic": false,
          "rules": Array [
            "
    box-shadow: 0 2px 20px 0 rgba(0,0,0,.1);
    border-radius: 100px;
    font-size: 12px;
    line-height: 15px;
    padding: .50em 10px;
    background-color: ",
            [Function],
            ";
    color: #000000;
    margin: 0 2px;
    text-decoration: none!important;
    &:hover {
      text-decoration: none!important;
      color: #000000;
    }
",
          ],
        },
        "displayName": "StoreLabelWrapper",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-ifAKCX",
        "target": "span",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
